pkg/placement/placement-resolver.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package placement
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/kubestellar/kubestellar/api/edge/v1alpha1"
+	"github.com/kubestellar/kubestellar/pkg/util"
+)
+
+// A PlacementResolver holds a collection of placement resolutions.
+// The collection is indexed by placementKey string, the resolver does not
+// care what the strings are, but a string should be a placement's unique
+// identifier. The resolution for a given key can be updated,
+// exported and compared to the PlacementDecision representation.
+type PlacementResolver interface {
+	// GeneratePlacementDecision returns the placement decision for the given
+	// placement key. If a key does not exist in the cache, it is created.
+	// This function can fail if any object's GVR fails to be retrieved given
+	// its GVK, by the mapper shared with the placement-controller.
```

Comment:
FYI: The definition of this interface does not say anything about a mapper shared with the placement-controller, does not even say anything about a placement-controller at all.

Reply:
I don't think it's the resolver's responsibility to dictate whether to retry or not. The resolver states what can cause errors, and the reader should decide - depending on his use - whether to retry or not. Is this not right?

Reply:
Yes, I see that is done adequately for the resolver.